File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap();
        Integer halfNumsSize = nums.length / 2 + 1;
        for (int num : nums) {
            Integer key = num;
            if (map.containsKey(key)) {
                Integer newV = map.get(key) + 1;
                if (halfNumsSize <= newV) return num;
                map.put(Integer.valueOf(num), newV);
            } else {
                Integer newV = 1;
                if (halfNumsSize <= newV) return num;
                map.put(key, 1);
            }
        }
        return -1;
    }
}

File: 0169-majority-element/0169-majority-element.kt
class Solution {
    fun majorityElement(nums: IntArray): Int {
        val halfNumsSize = nums.size / 2 + 1
        val map = HashMap<Int, Int>();
        for (num in nums) {
            val newV = map.getOrDefault(num, 0) + 1
            if (newV >= halfNumsSize) return num
            map.put(num, newV)
        }
        return -1
    }
}